### ISSUE TITLE
build_site() returning an error : Error in readRDS(path) : error reading from connection

### ISSUE THREAD

**Issue by xtimbeau** [r-lib/pkgdown#2713]
I am building vignettes using quarto as the vignette builder.

When building site, I have an error. It occurs in `build_home_index()` more specifically in the function `cran_link()`.
When commenting out line 209 of `build_home_index.R`, the error goes away and the building process completes.

No idea why it is occuring, nor why it is not occuring anymore. 

I can try to build a reprex, but that was not systematic in the last days, I suspect it started after some packages update.

```
cran_link <- function(pkg) {
  if (!has_internet()) {
    return(NULL)
  }

  cran_url <- paste0("https://cloud.r-project.org/package=", pkg)
  req <- httr2::request(cran_url)
  req <- req_pkgdown_cache(req)
  req <- httr2::req_error(req, function(resp) FALSE)
  resp <- httr2::req_perform(req)
  if (!httr2::resp_is_error(resp)) {
    return(list(repo = "CRAN", url = cran_url))
  }

  # bioconductor always returns a 200 status, redirecting to /removed-packages/
  bioc_url <- paste0("https://www.bioconductor.org/packages/", pkg)
  req <- httr2::request(bioc_url)
  # req <- req_pkgdown_cache(req)
  req <- httr2::req_error(req, function(resp) FALSE)
  req <- httr2::req_retry(req, max_tries = 3)
  resp <- httr2::req_perform(req)

  if (!httr2::resp_is_error(resp) && !grepl("removed-packages", httr2::resp_url(resp))) {
    return(list(repo = "Bioconductor", url = bioc_url))
  }

  NULL
}
```

**Comment by jayhesselberth** [comment]
Does it look like [this error](https://github.com/r-lib/pkgdown/issues/2696)?

You might try installing the dev httr2.

**Comment by jayhesselberth** [comment]
This is also happening in the r-devel-linux automated [CRAN checks](https://cran.r-project.org/web/checks/check_results_pkgdown.html).

**Comment by remlapmot** [comment]
This has also happened to me. I am also not sure why it happened, but I noted <https://zenodo.org/> was down the day it happened and I use a README badge from them. As noted in <https://github.com/r-lib/pkgdown/pull/2676#issuecomment-2195455780> I had to delete my pkgdown cache before I could successfully build a site again.

```r
fs::dir_delete(fs::path(tools::R_user_dir("pkgdown", "cache"), "http"))
```

**Comment by xtimbeau** [comment]
exactly the same
i’ll try httr2 dev

Le sam. 13 juil. 2024 à 00:32, Jay Hesselberth ***@***.***> a
écrit :

**Comment by jayhesselberth** [comment]
Can you try installing this branch and the re-build your site?

``` r
pak::pak("r-lib/pkgdown@cache-path")
```

**Comment by jayhesselberth** [comment]
Addressed in #2715

**Comment by xtimbeau** [comment]
it is working !

Le sam. 13 juil. 2024 à 15:18, Jay Hesselberth ***@***.***> a
écrit :
